api/grpcserver/smesher_service.go

```diff
@@ -78,21 +76,16 @@ func (s SmesherService) StopSmeshing(ctx context.Context, in *pb.StopSmeshingReq
 func (s SmesherService) SmesherID(ctx context.Context, in *empty.Empty) (*pb.SmesherIDResponse, error) {
 	log.Info("GRPC SmesherService.SmesherID")
 
-	smesherID := s.Mining.GetSmesherID()
-	return &pb.SmesherIDResponse{AccountId: &pb.AccountId{Address: smesherID.ToBytes()}}, nil
+	addr := util.Hex2Bytes(s.smeshing.SmesherID().Key)
+	return &pb.SmesherIDResponse{AccountId: &pb.AccountId{Address: addr}}, nil
 }
 
 // Coinbase returns the current coinbase setting of this node
 func (s SmesherService) Coinbase(ctx context.Context, in *empty.Empty) (*pb.CoinbaseResponse, error) {
 	log.Info("GRPC SmesherService.Coinbase")
 
-	_, _, coinbase, _ := s.Mining.MiningStats()
-	addr, err := types.StringToAddress(coinbase)
-	if err != nil {
-		log.Error("error converting coinbase: %s", err)
-		return nil, status.Errorf(codes.Internal, "error reading coinbase data")
-	}
-	return &pb.CoinbaseResponse{AccountId: &pb.AccountId{Address: addr.Bytes()}}, nil
+	addr := s.smeshing.Coinbase()
+	return &pb.CoinbaseResponse{AccountId: &pb.AccountId{Address: addr[:]}}, nil
```

Comment:
For my own edification, what does `addr[:]` do? Is it the same as `addr.Bytes()`?

Reply:
Yes, it's the same. I guess I changed it after I rewrote this section, and didn't notice the `Bytes()` alias, which is probably preferable. Fixed.